File: react-native/__tests__/dittoMode.test.ts
import {selectMode} from '../dittoMode';

describe('selectMode', () => {
  it('null token selects online', () => {
    expect(selectMode(null)).toBe('online');
  });

  it('undefined token selects online', () => {
    expect(selectMode(undefined)).toBe('online');
  });

  it('empty token selects online', () => {
    expect(selectMode('')).toBe('online');
  });

  it('whitespace-only token selects online', () => {
    expect(selectMode('   \t\n  ')).toBe('online');
  });

  it('non-empty token selects offline', () => {
    expect(selectMode('any-real-license-token')).toBe('offline');
  });
});

File: flutter_app/test/ditto_mode_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:flutter_quickstart/ditto_mode.dart';

void main() {
  group('selectDittoMode', () {
    test('null token selects online', () {
      expect(selectDittoMode(null), DittoMode.onlinePlayground);
    });

    test('empty token selects online', () {
      expect(selectDittoMode(''), DittoMode.onlinePlayground);
    });

    test('whitespace-only token selects online', () {
      expect(selectDittoMode('   \t\n  '), DittoMode.onlinePlayground);
    });

    test('non-empty token selects offline', () {
      expect(selectDittoMode('any-real-license-token'), DittoMode.offline);
    });
  });
}
